src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+import duke.command.DukeException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.Scanner;
+
 public class Duke {
+    /**
+     * Executes the "Duke" program.
+     *
+     * @param args The command line arguments.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        //Initialisation
+        String line = "start";
+        Task[] list = new Task[100];
+        int listSize = 0;
+        Scanner input = new Scanner(System.in);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        //Get commands while user input is not "bye".
+        while (!line.equals("bye")) {
+            line = input.nextLine();
+            try {
+                listSize = handleUserInputs(line, list, listSize);
+            } catch (DukeException de) {
+                handleException(line);
+            }
+        }
     }
+
+    /**
+     * Returns the number of tasks found in the list, after handling the commands given by the
+     * user in the form of a line of string.
+     *
+     * @param line     The single line of string inputted by the user.
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @return Returns the current size of the list populated by tasks created by the user.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static int handleUserInputs(String line, Task[] list, int listSize) throws DukeException {
+        String[] command = line.split(" ");
+        //Check the first word in the line of strings (list, mark, unmark, event, etc.)
+        switch (command[0]) {
+        case "list":
+            printList(list, listSize, command);
+            break;
+        case "mark":
+            markAsDone(list, listSize, command);
+            break;
+        case "unmark":
+            markAsUndone(list, listSize, command);
+            break;
+        case "todo":
+            if (command.length == 1) {
+                throw new DukeException();
+            }
+            list[listSize] = new ToDo(line.substring(line.indexOf(' ') + 1));
```

Comment:
Try not to use magic numbers like `line.indexOf(' ') + 1` throughout your code which will make it confusing to the reader what you are referring to